diff --git a/node_modules/@material/mwc-textfield/mwc-textfield-base.js b/node_modules/@material/mwc-textfield/mwc-textfield-base.js
index 5d17fac..f827a0e 100644
--- a/node_modules/@material/mwc-textfield/mwc-textfield-base.js
+++ b/node_modules/@material/mwc-textfield/mwc-textfield-base.js
@@ -208,7 +208,7 @@ export class TextFieldBase extends FormElement {
             'mdc-text-field__icon--leading': !isTrailingIcon,
             'mdc-text-field__icon--trailing': isTrailingIcon
         };
-        return html `<i class="material-icons mdc-text-field__icon ${classMap(classes)}">${icon}</i>`;
+        return html `<i class="material-icons mdc-text-field__icon ${classMap(classes)}" style="color:var(--general-colorTextSecondary);">${icon}</i>`;
     }
     /** @soyTemplate */
     renderPrefix() {
@@ -225,7 +225,7 @@ export class TextFieldBase extends FormElement {
             'mdc-text-field__affix--prefix': !isSuffix,
             'mdc-text-field__affix--suffix': isSuffix
         };
-        return html `<span class="mdc-text-field__affix ${classMap(classes)}">
+        return html `<span class="mdc-text-field__affix ${classMap(classes)}" style="${this.disabled?"color:var(--general-colorTextDisabled);":"color:var(--general-colorTextSecondary);"}">
         ${content}</span>`;
     }
     /** @soyTemplate */
