diff --git a/node_modules/@material/mwc-top-app-bar/mwc-top-app-bar-base-base.js b/node_modules/@material/mwc-top-app-bar/mwc-top-app-bar-base-base.js
index 80d351b..de15325 100644
--- a/node_modules/@material/mwc-top-app-bar/mwc-top-app-bar-base-base.js
+++ b/node_modules/@material/mwc-top-app-bar/mwc-top-app-bar-base-base.js
@@ -50,6 +50,28 @@ export class TopAppBarBaseBase extends BaseElement {
         }
         // clang-format on
         return html `
+        <style>
+         .mdc-top-app-bar {
+            width: calc(100% - var(--mdc-drawer-width, 256px));
+            }
+            .mdc-top-app-bar__title {
+            padding-left: 0;
+            }
+            .mdc-top-app-bar--prominent .mdc-top-app-bar__row {
+            height: 96px;
+            }
+            .mdc-top-app-bar--prominent-fixed-adjust {
+            padding-top: 96px;
+            }
+
+            /* min-width for mobile-view */
+            @media screen and (max-width: 375px) {
+            .mdc-top-app-bar {
+                min-width: 375px !important;
+            }
+        }
+        </style>
+    <div class="mdc-top-app-bar--fixed-adjust"></div>
       <header class="mdc-top-app-bar ${classMap(this.barClasses())}">
       <div class="mdc-top-app-bar__row">
         <section class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start" id="navigation">
