diff --git a/node_modules/@material/mwc-drawer/mwc-drawer-base.js b/node_modules/@material/mwc-drawer/mwc-drawer-base.js
index 87857a7..da36d3a 100644
--- a/node_modules/@material/mwc-drawer/mwc-drawer-base.js
+++ b/node_modules/@material/mwc-drawer/mwc-drawer-base.js
@@ -79,6 +79,25 @@ export class DrawerBase extends BaseElement {
             'mdc-drawer--modal': modal,
         };
         return html `
+      <style>
+        .mdc-drawer {
+          background-color: var(--mdc-drawer-background-color, '#ffffff');
+          width: var(--mdc-drawer-width, 256px);
+          border-left: var(
+            --mdc-drawer-border-left,
+            1px solid rgba(0, 0, 0, 0.12)
+          );
+          border-right: var(
+            --mdc-drawer-border-right,
+            1px solid rgba(0, 0, 0, 0.12)
+          );
+        }
+        .mdc-drawer.mdc-drawer--open:not(.mdc-drawer--closing)
+          + .mdc-drawer-app-content {
+          margin-left: var(--mdc-drawer-width, 256px) !important;
+          width: calc(100% - var(--mdc-drawer-width, 256px));
+        }
+      </style>
       <aside class="mdc-drawer ${classMap(classes)}">
         ${header}
         <div class="mdc-drawer__content"><slot></slot></div>
