diff --git a/node_modules/@material/mwc-snackbar/mwc-snackbar-base.js b/node_modules/@material/mwc-snackbar/mwc-snackbar-base.js
index 3ba93ac..9d52adf 100644
--- a/node_modules/@material/mwc-snackbar/mwc-snackbar-base.js
+++ b/node_modules/@material/mwc-snackbar/mwc-snackbar-base.js
@@ -32,6 +32,16 @@ export class SnackbarBase extends BaseElement {
             'mdc-snackbar--leading': this.leading,
         };
         return html `
+      <style>
+        .mdc-snackbar {
+          position: fixed;
+          bottom: var(--mdc-snackbar-bottom, 20px);
+        }
+        .mdc-snackbar__surface {
+          position: absolute;
+          right: 20px;
+        }
+      </style>
       <div class="mdc-snackbar ${classMap(classes)}" @keydown="${this._handleKeydown}">
         <div class="mdc-snackbar__surface">
           ${accessibleSnackbarLabel(this.labelText, this.open)}
