diff --git a/node_modules/@material/mwc-dialog/mwc-dialog-base.js b/node_modules/@material/mwc-dialog/mwc-dialog-base.js
index a2ec1b3..5e64074 100644
--- a/node_modules/@material/mwc-dialog/mwc-dialog-base.js
+++ b/node_modules/@material/mwc-dialog/mwc-dialog-base.js
@@ -179,15 +179,25 @@ export class DialogBase extends BaseElement {
             'mdc-dialog__actions': !this.hideActions,
         };
         return html `
+    <style>
+        .mdc-dialog .mdc-dialog__content {
+          padding: 0;
+        }
+        .mdc-dialog .mdc-dialog__container,
+        .mdc-dialog .mdc-dialog__surface {
+          height: var(--mdc-dialog-height, auto) !important;
+          width: var(--mdc-dialog-width, auto) !important;
+        }
+    </style>
     <div class="mdc-dialog ${classMap(classes)}"
         role="alertdialog"
         aria-modal="true"
         aria-labelledby="title"
         aria-describedby="content">
-      <div class="mdc-dialog__container">
-        <div class="mdc-dialog__surface">
+      <div class="mdc-dialog__container" style="height: var(--mdc-dialog-height, auto) !important;width: var(--mdc-dialog-width, auto) !important;">
+        <div class="mdc-dialog__surface" style="height: var(--mdc-dialog-height, auto) !important;width: var(--mdc-dialog-width, auto) !important;">
           ${heading}
-          <div id="content" class="mdc-dialog__content">
+          <div id="content" class="mdc-dialog__content" style="padding:0;">
             <slot id="contentSlot"></slot>
           </div>
           <footer
