--- ./mwc-select-base.ts	2020-08-10 17:10:24.000000000 +0900
+++ ./mwc-multi-select-base.ts	2020-08-10 17:11:20.000000000 +0900
@@ -181,6 +181,8 @@
 
   @property({type: Boolean}) naturalMenuWidth = false;
 
+  @property({type: Boolean}) multi = false;
+
   @property({type: Boolean}) protected isUiValid = true;
 
   // Transiently holds current typeahead prefix from user.
@@ -315,6 +317,7 @@
             class="mdc-select__menu mdc-menu mdc-menu-surface ${
         classMap(menuClasses)}"
             activatable
+            .multi=${this.multi}
             .fullwidth=${!this.naturalMenuWidth}
             .open=${this.menuOpen}
             .anchor=${this.anchorElement}
@@ -418,6 +421,7 @@
         return !!this.label;
       },
       floatLabel: (shouldFloat) => {
+        shouldFloat = this.selectedText !== '' ? true : false; // this is a temporary change for label floating.
         if (this.labelElement) {
           this.labelElement.floatingLabelFoundation.float(shouldFloat);
         }
@@ -454,7 +458,14 @@
         }
       },
       notifyChange: async (value) => {
-        if (!this.valueSetDirectly && value === this.value) {
+        if (!this.valueSetDirectly && this.multi && this.menuElement !== null) {
+          let values: string[] = [];
+          (this.menuElement.selected as any).forEach(item => {
+            values.push(item.value);
+          });
+          this.selectedText = values.join(',');
+          //this.value = values;
+        } else if (!this.valueSetDirectly && value === this.value) {
           return;
         }
 
@@ -465,6 +476,7 @@
         this.dispatchEvent(ev);
       },
       setSelectedText: (value) => this.selectedText = value,
+      // @ts-ignore
       isSelectAnchorFocused: () => {
         const selectAnchorElement = this.anchorElement;
 
@@ -709,8 +721,11 @@
     }
   }
 
-  protected selectByValue(value: string) {
+  protected selectByValue(value: any) {
     let indexToSelect = -1;
+    if (this.multi) {
+      return;
+    }
     for (let i = 0; i < this.items.length; i++) {
       const item = this.items[i];
       if (item.value === value) {
@@ -843,9 +858,19 @@
     }
 
     this.mdcFoundation.handleMenuItemAction(event.detail.index);
-    const item = this.items[event.detail.index];
-    if (item) {
-      this.value = item.value;
+    // CHANGED : to provide multiple selection
+    if (this.multi) {
+      if (this.selected === null) {
+        this.value = '';
+      } else {
+        this.value = (this.selected as any).map(a => a.value);
+        this.selectedText = this.value.toString();
+      }
+    } else {
+      const item = this.items[event.detail.index];
+      if (item) {
+        this.value = item.value;
+      }
     }
   }
 
