--- ./mwc-select-base.ts	2020-05-08 10:45:49.000000000 +0900
+++ ./mwc-multi-select-base.ts	2020-05-08 10:44:13.000000000 +0900
@@ -168,6 +168,8 @@
 
   @property({type: Boolean}) naturalMenuWidth = false;
 
+  @property({type: Boolean}) multi = false;
+
   @property({type: Boolean}) protected isUiValid = true;
 
   get items(): ListItemBase[] {
@@ -294,6 +296,7 @@
             wrapFocus
             class="mdc-select__menu mdc-menu mdc-menu-surface"
             activatable
+            .multi=${this.multi}
             .fullwidth=${!this.naturalMenuWidth}
             .open=${this.menuOpen}
             .anchor=${this.anchorElement}
@@ -387,6 +390,7 @@
         return !!this.label;
       },
       floatLabel: (shouldFloat) => {
+        shouldFloat = this.selectedText !== '' ? true : false; // this is a temporary change for label floating.
         if (this.labelElement) {
           this.labelElement.floatingLabelFoundation.float(shouldFloat);
         }
@@ -418,7 +422,14 @@
         }
       },
       notifyChange: async (value) => {
-        if (!this.valueSetDirectly && value === this.value) {
+        if (!this.valueSetDirectly && this.multi && this.menuElement !== null) {
+          let values: string[] = [];
+          (this.menuElement.selected as any).forEach(item => {
+            values.push(item.value);
+          });
+          this.selectedText = values.join(',');
+          //this.value = values;
+        } else if (!this.valueSetDirectly && value === this.value) {
           return;
         }
 
@@ -429,6 +440,7 @@
         this.dispatchEvent(ev);
       },
       setSelectedText: (value) => this.selectedText = value,
+      // @ts-ignore
       isSelectAnchorFocused: () => {
         const selectAnchorElement = this.anchorElement;
 
@@ -626,8 +638,11 @@
     }
   }
 
-  protected selectByValue(value: string) {
+  protected selectByValue(value: any) {
     let indexToSelect = -1;
+    if (this.multi) {
+      return;
+    }
     for (let i = 0; i < this.items.length; i++) {
       const item = this.items[i];
       if (item.value === value) {
@@ -715,6 +730,15 @@
     }
 
     this.mdcFoundation.handleMenuItemAction(evt.detail.index);
+    // CHANGED : to provide multiple selection
+    if (this.multi) {
+      if (this.selected === null) {
+        this.value = '';
+      } else {
+        this.value = (this.selected as any).map(a => a.value);
+        this.selectedText = this.value.toString();
+      }
+    }
   }
 
   protected onOpened() {
