--- ./mwc-select-base.ts	2020-07-01 19:53:11.000000000 +0900
+++ ./mwc-multi-select-base.ts	2020-07-01 19:56:01.000000000 +0900
@@ -180,6 +180,8 @@
 
   @property({type: Boolean}) naturalMenuWidth = false;
 
+  @property({type: Boolean}) multi = false;
+
   @property({type: Boolean}) protected isUiValid = true;
 
   // Transiently holds current typeahead prefix from user.
@@ -320,6 +322,7 @@
             class="mdc-select__menu mdc-menu mdc-menu-surface ${
         classMap(menuClasses)}"
             activatable
+            .multi=${this.multi}
             .fullwidth=${!this.naturalMenuWidth}
             .open=${this.menuOpen}
             .anchor=${this.anchorElement}
@@ -413,6 +416,7 @@
         return !!this.label;
       },
       floatLabel: (shouldFloat) => {
+        shouldFloat = this.selectedText !== '' ? true : false; // this is a temporary change for label floating.
         if (this.labelElement) {
           this.labelElement.floatingLabelFoundation.float(shouldFloat);
         }
@@ -449,7 +453,14 @@
         }
       },
       notifyChange: async (value) => {
-        if (!this.valueSetDirectly && value === this.value) {
+        if (!this.valueSetDirectly && this.multi && this.menuElement !== null) {
+          let values: string[] = [];
+          (this.menuElement.selected as any).forEach(item => {
+            values.push(item.value);
+          });
+          this.selectedText = values.join(',');
+          //this.value = values;
+        } else if (!this.valueSetDirectly && value === this.value) {
           return;
         }
 
@@ -460,6 +471,7 @@
         this.dispatchEvent(ev);
       },
       setSelectedText: (value) => this.selectedText = value,
+      // @ts-ignore
       isSelectAnchorFocused: () => {
         const selectAnchorElement = this.anchorElement;
 
@@ -711,8 +723,11 @@
     }
   }
 
-  protected selectByValue(value: string) {
+  protected selectByValue(value: any) {
     let indexToSelect = -1;
+    if (this.multi) {
+      return;
+    }
     for (let i = 0; i < this.items.length; i++) {
       const item = this.items[i];
       if (item.value === value) {
@@ -845,6 +860,15 @@
     }
 
     this.mdcFoundation.handleMenuItemAction(evt.detail.index);
+    // CHANGED : to provide multiple selection
+    if (this.multi) {
+      if (this.selected === null) {
+        this.value = '';
+      } else {
+        this.value = (this.selected as any).map(a => a.value);
+        this.selectedText = this.value.toString();
+      }
+    }
   }
 
   protected onOpened() {
